pkg/config/utils/miscellaneous.go

```diff
@@ -31,3 +31,21 @@ func SetLogLevel(level string, config pkgconfigmodel.Writer, source pkgconfigmod
 	config.Set("log_level", seelogLogLevel, source)
 	return nil
 }
+
+// IsCoreAgentEnabled checks if the Agent is able to send the payloads it and other Agents need to function with
+func IsCoreAgentEnabled(cfg pkgconfigmodel.Reader) bool {
```

Comment:
nit: IMO this function name is bit deceptive, I would have expected only `core_agent.enabled` to be relevant for it's resolution